pyproject.toml

```diff
@@ -1,6 +1,58 @@
 # Copyright 2021 Canonical Ltd.
 # See LICENSE file for licensing details.
 
+[tool.poetry]
+name = "postgesql-k8s"
+version = "0.0.1-dev.0"
+description = ""
+authors = []
+license = "Apache-2.0"
+readme = "README.md"
+homepage = "https://charmhub.io/postgresql-k8s"
+repository = "https://github.com/canonical/postgresql-k8s-operator"
+
+[tool.poetry.dependencies]
+python = "^3.8.10"
+ops = "^2.0"
+pgconnstr = "^1.0.1"
+tenacity = "^8.1.0"
+cryptography = "^39.0.0"
+jsonschema = "^4.17.3"
+jinja2 = "^3.1.2"
+lightkube = "^0.12.0"
+lightkube-models = "^1.26.0.4"
+requests = "^2.28.2"
+# psycopg2 = "^2.9.5" # Injected in charmcraft.yaml
+
+[tool.poetry.group.fmt.dependencies]
+black = "^22.12.0"
+isort = "^5.11.4"
+
+[tool.poetry.group.lint.dependencies]
+flake8 = "^5.0.4"
+flake8-docstrings = "^1.7.0"
+flake8-copyright = "^0.2.3"
+flake8-builtins = "^2.1.0"
+pyproject-flake8 = "^5.0.4.post1"
+pep8-naming = "^0.13.3"
+codespell = "^2.2.2"
+
+[tool.poetry.group.unit.dependencies]
+coverage = {extras = ["toml"], version = "^7.0.5"}
+pytest = "^7.1.1"
+pytest-asyncio = "^0.20.3"
+
+[tool.poetry.group.integration.dependencies]
+lightkube = "^0.12.0"
+pytest = "^7.1.1"
+pytest-operator = "^0.23.0"
+juju = "^2" # Latest juju 2
+psycopg2 = {version = "^2.9.5", extras = ["binary"]}
```

Comment:
Corresponds to environment deps in `tox.ini`. Poetry will pull them only if requested with `--with`